### PR TITLE
Inverse Distance Weighting Interpolator

### PR DESCRIPTION
**Pull request by mcara**
Add a module for performing Inverse Distance Weighted (IDW) interpolation on unstructured data using a modified version of the [Shepard's method](http://en.wikipedia.org/wiki/Inverse_distance_weighting). This module (idw) uses `scipy.spatial.cKDTree` to retrieve nearest neighbors for distance weighting.

### CODE REVIEW

File: imutils/idw.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+__vdate__ = '2015-04-25'
+__author__ = "Mihai Cara"
+
+
+__all__ = [ 'ShepardIDWInterpolator' ]
+
+
+class ShepardIDWInterpolator(object):
+    """
+Class to perform Inverse Distance Weighted (IDW) interpolation on unstructured
```
Comment (larrybradley) [inline imutils/idw.py]: The docstrings should be indented to the same level as `"""`.

File: imutils/idw.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+data using a modified version of the
+`Shepard's method <http://en.wikipedia.org/wiki/Inverse_distance_weighting>`_
+(see Notes section for details).
+
+Examples
```
Comment (larrybradley) [inline imutils/idw.py]: The Examples section should go after the Notes section.

File: imutils/idw.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        self.kdtree = cKDTree(coord, leafsize=leafsize)
+        self.ncoord_dim = ncoord_dim
+        self.ncoord = ncoord
+        self.vals = vals
+        self.weights = weights
+
+
+    def __call__(self, pts, nbr=8, eps=0.0, p=1, reg=0.0, confdist=1e-12, dtype=np.float):
+        """
+__call__(self, pts, nbr=8, eps=0.0, p=1, reg=0.0, confdist=1e-12, dtype=np.float)
```
Comment (larrybradley) [inline imutils/idw.py]: indent docstring

**Comment by mcara** [comment]
Cannot solve merge conflicts. Will try again later.

**Pull request closed without merging**